tests/org.eclipse.ui.tests.harness/src/org/eclipse/ui/tests/harness/util/RCPTestWorkbenchAdvisor.java

```diff
@@ -199,6 +218,22 @@ public void run() {
 	@Override
 	public void postStartup() {
 		super.postStartup();
+
+		// Wait for all async/sync operations to complete before marking as started
+		// This prevents a race condition where asyncWithDisplayAccess might not be set yet
+		if (asyncLatch != null) {
+			try {
+				// Wait up to 5 seconds for all operations to complete
+				boolean completed = asyncLatch.await(5, java.util.concurrent.TimeUnit.SECONDS);
+				if (!completed) {
+					System.err.println("Warning: Not all async/sync operations completed within timeout");
+				}
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				System.err.println("Warning: Interrupted while waiting for async/sync operations");
```

Comment:
It's probably safer to simply re-throw the exception and let it fail too. Otherwise the test may fail afterwards and one will have to look into the logs to see why.
```suggestion
throw e;
```